operator/pkg/gateway-api/gateway_status.go

```diff
@@ -45,6 +51,29 @@ func gatewayStatusAcceptedCondition(gw *gatewayv1beta1.Gateway, accepted bool, m
 	}
 }
 
+func gatewayStatusProgrammedCondition(gw *gatewayv1beta1.Gateway, scheduled bool, msg string) metav1.Condition {
+	switch scheduled {
```

Comment:
Just wondering, why `switch` not `if`?

Reply:
This is an adaptation from @sayboras's original code, not sure what his reasoning was :)

Reply:
ah i might have started with some enum, and then refactored to bool, so switch might be just lazy refactor.

Reply:
I have had an idea last night to re-factor this into 1 generic blob to re-use. But that is probably out of scope for this PR.